### PR TITLE
git-sync: added ssh to path

### PR DESCRIPTION
**Pull request by cab404**
### Description
This fixed automated pushing to repository

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [X] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [X] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by cab404** [comment]
wat. @rycee?

**Comment by cab404** [comment]
Let's just say the test is stupid and stiffling

**Comment by cab404** [comment]
Well, that's not me

**Comment by cab404** [comment]
@rycee please <3

File: tests/modules/services/git-sync/basic.nix [hunk 1]
```diff
@@ -15,23 +15,13 @@
...
-          Description=Git Sync test
-        ''
-      }
+      assertFileContains $serviceFile "Environment=GIT_SYNC_DIRECTORY=/a/path";
+      assertFileContains $serviceFile "Environment=GIT_SYNC_COMMAND=@git-sync@/bin/git-sync"
+      assertFileContains $serviceFile "GIT_SYNC_REPOSITORY=git+ssh://user@example.com:/~user/path/to/repo.git"
+      assertFileContains $serviceFile "Environment=${
+        builtins.elemAt
+        config.systemd.user.services.git-sync-test.Service.Environment 0
+      }"
```
Comment (sumnerevans) [inline tests/modules/services/git-sync/basic.nix:24]: Why this change? You can use `normalizeStorePaths` if you are concerned about the bin path changing.

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: tests/modules/services/git-sync/basic.nix (hunk 1 above)
Comment (cab404) [reply tests/modules/services/git-sync/basic.nix:24]: I am weirded out that each change to service needs to be 1-1 changed in the tests — it's just plain weird to make it false-positive any new addition as an error

Comment (cab404) [reply tests/modules/services/git-sync/basic.nix]: but ok

Comment (sumnerevans) [reply tests/modules/services/git-sync/basic.nix]: That is the entire point of the tests. We want to ensure that every change to the output is intentional, not an accident.

Comment (cab404) [reply tests/modules/services/git-sync/basic.nix]: well, I fixed it

**Comment by cab404** [comment]
thank you!
